### ISSUE TITLE
Deploying to Vercel requires symlinks to have absolute paths for internal packages

### ISSUE THREAD

**Issue by mikestopcontinues** [pnpm/pnpm#3601]
### Code to reproduce the issue:

That's not an easy thing since it's platform-dependent. The [issue](https://github.com/vercel/vercel/issues/2432) is well-known. It seems that internally linked packages must have absolute paths to work.

I tried using `npm i --package-import-method copy`, and it didn't fix the internal package problem (though I assume it did copy packages from outside the repo). I've gotten many variations of the error shown below (for instance, if I try to run tests, my shared preset config throws the error).

At first glance, it seems like an additional config option for absolute links would solve the problem, but I honestly only switched to pnpm today because I was getting the same error with npm workspaces. I tried yarn too, which also gives the error.

### Expected behavior:

I would like to deploy a monorepo with shared code to Vercel.

### Actual behavior:

```
17:30:53.513 | > Build error occurred
-- | --
17:30:53.515 | Error: Cannot find module '/vercel/path0/app/app-view/node_modules/@arc/is-env/dist/index.js'
17:30:53.515 | at createEsmNotFoundErr (internal/modules/cjs/loader.js:929:15)
17:30:53.515 | at finalizeEsmResolution (internal/modules/cjs/loader.js:922:15)
17:30:53.515 | at resolveExports (internal/modules/cjs/loader.js:450:14)
17:30:53.515 | at Function.Module._findPath (internal/modules/cjs/loader.js:490:31)
17:30:53.515 | at Function.Module._resolveFilename (internal/modules/cjs/loader.js:888:27)
17:30:53.515 | at Function.mod._resolveFilename (/vercel/path0/node_modules/.pnpm/next@11.0.1_f31b35bdaa3b7c2fce82404f09d2dac4/node_modules/next/dist/build/webpack/require-hook.js:4:1855)
17:30:53.515 | at Function.Module._load (internal/modules/cjs/loader.js:746:27)
17:30:53.515 | at Module.require (internal/modules/cjs/loader.js:974:19)
17:30:53.515 | at require (internal/modules/cjs/helpers.js:92:18)
17:30:53.515 | at Object.<anonymous> (/vercel/path0/app/app-view/next.config.__vercel_builder_backup__.js:8:51) {
17:30:53.515 | code: 'MODULE_NOT_FOUND',
17:30:53.515 | path: '/vercel/path0/app/app-view/node_modules/@arc/is-env/package.json'
17:30:53.515 | }
17:30:53.522 | ERROR  Command failed with exit code 1.
17:30:53.537 | Error: Command "pnpm run build" exited with 1
```

### Additional information:

 - `node -v` prints: 14.10.1
 - Windows, macOS, or Linux?: I'm pretty sure Vercel builds in a NodeJS lambda on AWS.

Thanks for any help you can offer!

**Comment by zkochan** [comment]
I think it is OK to add an option for this. By the way, on Windows pnpm uses junctions, which use absolute paths.

**Comment by mikestopcontinues** [comment]
@zkochan Hey! I just got word from Vercel that the problem is (no longer?) relative paths. It seems they don't support symlinks at all. That ticket is from when the company was still called Now, so I think they must've migrated since then. I also ran a quick test this morning rewriting the links to absolute, and can confirm it definitely doesn't work anymore.

Anyway, thanks for your time. At the very least, I'm happy it got me to switch over. `pnpm` is really nice!
